cassiopeia/data/CassiopeiaTree.py

```diff
@@ -645,9 +737,34 @@ def set_branch_length(self, parent: str, child: str, length: float):
         if length < 0:
             raise CassiopeiaTreeError("Edge length must be positive.")
 
-        self.__network[parent][child]["length"] = length
+        self.__set_branch_length(parent, child, length)
 
-        for (u, v) in self.depth_first_traverse_edges(source=parent):
+        for u, v in self.depth_first_traverse_edges(source=parent):
+            self.__network.nodes[v]["time"] = (
+                self.__network.nodes[u]["time"] + self.__network[u][v]["length"]
+            )
+
+    def set_branch_lengths(self, branch_length_dict: Dict[Tuple[str, str], float]) -> None:
```

Comment:
Some places you have return `None`, some places you do not. Either is fine (I normally omit them), but can we make this consistent?